include/upipe/ubuf_pic.h

```diff
@@ -388,7 +397,69 @@ static inline int ubuf_pic_blit(struct ubuf *dest, struct ubuf *src,
         int plane_vsize = extract_vsize / src_vsub;
 
         for (int i = 0; i < plane_vsize; i++) {
-            memcpy(dest_buffer, src_buffer, plane_hsize);
+            if (!alpha_plane || threshold == 0) {
+                memcpy(dest_buffer, src_buffer, plane_hsize);
+            } else if (threshold != 0xff) {
+                for (int j = 0; j < plane_hsize; j += 16) {
+                    /* This is an on/off blending
+                     * if alpha is over the threshold, we use the subpicture pixel.
+                     */
+                    const uint8_t *a = &alpha_plane[alpha_stride * (i * src_vsub) + j * src_hsub];
+                    if (a[0x0] > threshold) dest_buffer[j+0x0] = src_buffer[j+0x0];
```

Comment:
benchmarking shows that at -O2 unrolling 4 loops is 4 times faster, unrolling more loops has less of an effect and can even be worse at -O3.

I will add tests to that commit log.

Also unrolling only helps for the threshold method, it doesn't do much for the division method, so it can be removed for that one